packages/stable/src/__tests__/api/documentsApi.int.spec.ts

```diff
@@ -3,12 +3,55 @@
 import CogniteClient from '../../cogniteClient';
 import { setupLoggedInClient } from '../testUtils';
 import { DocumentSearchResponse } from '@cognite/sdk-stable/dist';
+import { TextEncoder } from 'util';
+
+const getFileId = async (
+  client: CogniteClient,
+  deadline: number = 15000
+): Promise<number> => {
+  try {
+    const resp = await client.documents.list({
+      limit: 1,
+    });
+    return resp.items[0].id;
+  } catch (error) {
+    const fileContent = new TextEncoder().encode('test data');
+    await client.files.upload(
+      {
+        name: 'test.txt',
+        mimeType: 'text/plain',
+      },
+      fileContent,
+      false,
+      true
+    );
+  }
+
+  while (deadline > 0) {
+    try {
+      const resp = await client.documents.list({
+        limit: 1,
+      });
+      return resp.items[0].id;
+    } catch (error) {
+      const timeout = 3000;
+      await new Promise((resolve) => setTimeout(resolve, deadline));
```

Comment:
Did you intend to use `timeout` here?

Note that on the last iteration it will now sleep and then always fail since deadline will be `<= 0`

```suggestion
      await new Promise((resolve) => setTimeout(resolve, timeout));
```

Can we also check `error` properly? We should only catch the not found errors, and not e.g. network errors.

Reply:
oh, good you spotted the use of deadline.